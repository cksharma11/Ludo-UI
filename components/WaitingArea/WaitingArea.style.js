import css from 'styled-jsx/css';

const WaitingAreaStyles = css`
  .waiting-area {
    display: flex;
    flex-direction: column;
    height: 720px;
    width: 100%;
    align-items: center;
    padding: 20px;
  }

  .gameid {
    font-size: 40px;
    text-align: center;
    color: black;
  }

  .player-info {
    font-size: 20px;
    height: 50px;
    text-align: center;
  }

  ul {
    padding: 0;
    font-size: 30px;
    text-align: center;
    color: black;
    width: 300px;
    border-radius: 2px;
    border: 1px solid black;
  }
`;

export default WaitingAreaStyles;
