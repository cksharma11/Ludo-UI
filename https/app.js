import fetch from 'isomorphic-unfetch';

const makeCall = async (url, options) => {
  return fetch(url, options).then((res) => res);
};

const app = {
  get: async (url, options = {}) => {
    return makeCall(url, { ...options, method: 'GET' });
  },
  post: async (url, options = {}) => {
    return makeCall(url, { ...options, method: 'POST' });
  }
};

export default app;
