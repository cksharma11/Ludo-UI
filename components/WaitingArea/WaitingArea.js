import React, { useState } from 'react';
import PropTypes from 'prop-types';
import app from '../../https/app';
import Header from '../Header/Header';
import Loader from '../ui/Loader/Loader';
import WaitingAreaStyles from './WaitingArea.style';
import labels from '../../config/labels/labels';
import { API_URL } from '../../utils/utils';
import GlobalStyles from '../globalStyle';

const WaitingArea = ({ gameId }) => {
  const [players, setPlayers] = useState([]);

  const fetchPlayers = async () => {
    const gameData = await app.post(`${API_URL}/players`, {
      body: JSON.stringify({ gameId })
    });
    const { players: joinedPlayers, isStarted } = gameData;
    if (isStarted) {
      window.location.assign('/game');
    }
    setPlayers(joinedPlayers);
  };

  setTimeout(fetchPlayers, 1000);

  return (
    <div className="container">
      <Header title={labels.GAME_TITLE} />
      <div className="waiting-area">
        <p className="gameid">{`Game Id : ${gameId}`}</p>
        <ul className="players-info">
          {players.length &&
            players.map((player, index) => (
              <li key={player.playerId}>
                {index + 1}.{player.name}
                {labels.HAS_JOINED_LABEL}
              </li>
            ))}
        </ul>
        <Loader />
      </div>
      <style jsx>{WaitingAreaStyles}</style>
      <style jsx>{GlobalStyles}</style>
    </div>
  );
};

WaitingArea.propTypes = {
  gameId: PropTypes.oneOf([PropTypes.string, PropTypes.number])
};

WaitingArea.defaultProps = {
  gameId: -1
};
export default WaitingArea;
