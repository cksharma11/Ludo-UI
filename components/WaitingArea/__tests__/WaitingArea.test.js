import React from 'react';
import { shallow } from 'enzyme';
import app from '../../../https/app';
import WaitingArea from '../WaitingArea';

jest.mock('../../../https/app');
jest.useFakeTimers();

describe('WaitingArea', () => {
  it('should render properly', () => {
    const gameId = 1;
    const players = [{ name: 'jhon', playerId: 123 }];
    const wrapper = shallow(<WaitingArea gameId={gameId} players={players} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should render properly when game is not started', () => {
    app.post.mockImplementation(() =>
      Promise.resolve({
        joinedPlayers: [
          {
            playerName: 'dummyName',
            playerId: 'dummyId',
            color: 'red',
            coins: { 1: 0, 2: 0, 3: 0, 4: 0 }
          }
        ],
        isStarted: false
      })
    );

    const gameId = 1;
    const players = [{ name: 'jhon', playerId: 123 }];
    const wrapper = shallow(<WaitingArea gameId={gameId} players={players} />);
    expect(wrapper.debug()).toMatchSnapshot();
    jest.runAllTimers();
    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 1000);
  });

  it('should render properly when game is started', () => {
    window.location.assign = jest.fn();
    app.post.mockImplementation(() =>
      Promise.resolve({
        joinedPlayers: [
          {
            playerName: 'dummyName',
            playerId: 'dummyId',
            color: 'red',
            coins: { 1: 0, 2: 0, 3: 0, 4: 0 }
          }
        ],
        isStarted: true
      })
    );

    const gameId = 1;
    const players = [{ name: 'jhon', playerId: 123 }];
    const wrapper = shallow(<WaitingArea gameId={gameId} players={players} />);
    expect(wrapper.debug()).toMatchSnapshot();
    jest.runAllTimers();
    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 1000);
  });
});
