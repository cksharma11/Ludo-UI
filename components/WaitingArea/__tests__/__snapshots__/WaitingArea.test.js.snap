// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WaitingArea should render properly 1`] = `
"<Fragment>
  <Header title=\\"Ludo\\" />
  <div className=\\"jsx-2871010629 waiting-area\\">
    <p className=\\"jsx-2871010629 gameid\\">
      Game Id : 1
    </p>
    <ul className=\\"jsx-2871010629 players-info\\">
      <li className=\\"jsx-2871010629\\">
        1
        .
         has joined
      </li>
    </ul>
    <Loader />
    <JSXStyle id=\\"2871010629\\">
      .
      w
      a
      i
      t
      i
      n
      g
      -
      a
      r
      e
      a
      .
      j
      s
      x
      -
      2
      8
      7
      1
      0
      1
      0
      6
      2
      9
      {
      d
      i
      s
      p
      l
      a
      y
      :
      -
      w
      e
      b
      k
      i
      t
      -
      b
      o
      x
      ;
      d
      i
      s
      p
      l
      a
      y
      :
      -
      w
      e
      b
      k
      i
      t
      -
      f
      l
      e
      x
      ;
      d
      i
      s
      p
      l
      a
      y
      :
      -
      m
      s
      -
      f
      l
      e
      x
      b
      o
      x
      ;
      d
      i
      s
      p
      l
      a
      y
      :
      f
      l
      e
      x
      ;
      -
      w
      e
      b
      k
      i
      t
      -
      f
      l
      e
      x
      -
      d
      i
      r
      e
      c
      t
      i
      o
      n
      :
      c
      o
      l
      u
      m
      n
      ;
      -
      m
      s
      -
      f
      l
      e
      x
      -
      d
      i
      r
      e
      c
      t
      i
      o
      n
      :
      c
      o
      l
      u
      m
      n
      ;
      f
      l
      e
      x
      -
      d
      i
      r
      e
      c
      t
      i
      o
      n
      :
      c
      o
      l
      u
      m
      n
      ;
      h
      e
      i
      g
      h
      t
      :
      7
      2
      0
      p
      x
      ;
      w
      i
      d
      t
      h
      :
      1
      0
      0
      %
      ;
      b
      a
      c
      k
      g
      r
      o
      u
      n
      d
      :
      #
      d
      f
      d
      f
      d
      f
      ;
      -
      w
      e
      b
      k
      i
      t
      -
      a
      l
      i
      g
      n
      -
      i
      t
      e
      m
      s
      :
      c
      e
      n
      t
      e
      r
      ;
      -
      w
      e
      b
      k
      i
      t
      -
      b
      o
      x
      -
      a
      l
      i
      g
      n
      :
      c
      e
      n
      t
      e
      r
      ;
      -
      m
      s
      -
      f
      l
      e
      x
      -
      a
      l
      i
      g
      n
      :
      c
      e
      n
      t
      e
      r
      ;
      a
      l
      i
      g
      n
      -
      i
      t
      e
      m
      s
      :
      c
      e
      n
      t
      e
      r
      ;
      p
      a
      d
      d
      i
      n
      g
      :
      2
      0
      p
      x
      ;
      }
      .
      g
      a
      m
      e
      i
      d
      .
      j
      s
      x
      -
      2
      8
      7
      1
      0
      1
      0
      6
      2
      9
      {
      f
      o
      n
      t
      -
      s
      i
      z
      e
      :
      4
      0
      p
      x
      ;
      t
      e
      x
      t
      -
      a
      l
      i
      g
      n
      :
      c
      e
      n
      t
      e
      r
      ;
      c
      o
      l
      o
      r
      :
      b
      l
      a
      c
      k
      ;
      }
      .
      p
      l
      a
      y
      e
      r
      -
      i
      n
      f
      o
      .
      j
      s
      x
      -
      2
      8
      7
      1
      0
      1
      0
      6
      2
      9
      {
      f
      o
      n
      t
      -
      s
      i
      z
      e
      :
      2
      0
      p
      x
      ;
      h
      e
      i
      g
      h
      t
      :
      5
      0
      p
      x
      ;
      t
      e
      x
      t
      -
      a
      l
      i
      g
      n
      :
      c
      e
      n
      t
      e
      r
      ;
      }
      u
      l
      .
      j
      s
      x
      -
      2
      8
      7
      1
      0
      1
      0
      6
      2
      9
      {
      p
      a
      d
      d
      i
      n
      g
      :
      0
      ;
      f
      o
      n
      t
      -
      s
      i
      z
      e
      :
      3
      0
      p
      x
      ;
      t
      e
      x
      t
      -
      a
      l
      i
      g
      n
      :
      c
      e
      n
      t
      e
      r
      ;
      c
      o
      l
      o
      r
      :
      b
      l
      a
      c
      k
      ;
      w
      i
      d
      t
      h
      :
      3
      0
      0
      p
      x
      ;
      b
      o
      r
      d
      e
      r
      -
      r
      a
      d
      i
      u
      s
      :
      2
      p
      x
      ;
      b
      o
      r
      d
      e
      r
      :
      1
      p
      x
       
      s
      o
      l
      i
      d
       
      b
      l
      a
      c
      k
      ;
      }
    </JSXStyle>
  </div>
</Fragment>"
`;
